universe/Condition.cpp

```diff
@@ -5111,17 +5111,20 @@ namespace {
             if (!candidate)
                 return false;
 
-            const ShipDesign* design = nullptr;
+            std::shared_ptr<const Ship> ship = nullptr;
             if (auto fighter = std::dynamic_pointer_cast<const ::Fighter>(candidate)) {
                 // it is a fighter
-                design = Objects().Object<Ship>(fighter->LaunchedFrom())->Design();
-            } else if (auto ship = std::dynamic_pointer_cast<const ::Ship>(candidate)) {
-                // ir is a ship
-                design = ship->Design();
+                ship = std::move(Objects().Object<Ship>(fighter->LaunchedFrom()));
```

Comment:
don't think `std::move` makes sense here...

Reply:
cause ... its already an rvalue?

Reply:
`ObjectMap::Object<T>` returns a `shared_ptr<T>` by value, and that value doesn't have a name here, so yes, I think?

Reply:
I guess the same reasoning applies to the `std::dynamic_pointer_cast<const` ::Ship>(candidate)`.
The candidate is an lvalue, so I exptect the cast will return a new shared_ptr.

Reply:
If I get it right if `candidate` would be a rvalue the cast would  actually move starting with c++20 and using candidate after that line would become illegal.